src/decoder/mod.rs

```diff
@@ -167,9 +170,18 @@ impl<R: Read> Decoder<R> {
         self.limits = limits;
     }
 
+    /// Ignore parse text(text/ztxt/itxt) chunk while read meta data
+    pub fn ignore_text_chunk(&mut self) {
+        self.ignore_text_chunk = true;
+    }
+
     /// Reads all meta data until the first IDAT chunk
     pub fn read_info(self) -> Result<Reader<R>, DecodingError> {
-        let mut reader = Reader::new(self.r, StreamingDecoder::new(), self.transform, self.limits);
+        let mut decoder = StreamingDecoder::new();
+        if self.ignore_text_chunk {
+            decoder.ignore_text_chunk();
+        }
```

Comment:
This would also remove some control flow here (in particular we don't depend on it defaulting to `false` in streaming).